src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -1504,24 +1509,41 @@ window.qBittorrent.DynamicTable ??= (() => {
             return true;
         },
 
+        removeRow: function(rowId) {
+            this.selectedRows.erase(rowId);
+            this.rows.delete(rowId);
+            const tr = this.getTrByRowId(rowId);
+            tr?.destroy();
+        },
+
+        clear: function() {
+            this.deselectAll();
+            this.rows.clear();
+            const trs = this.tableBody.getElements("tr");
+            while (trs.length > 0)
+                trs.pop().destroy();
+        },
+
+        getRowIds: function() {
+            return [...this.rows.keys()];
+        },
```

Comment:
It seems that all existing usage of this function is doing: `torrentsTable.getRowIds().length;` which is not efficient with your implementation.
If possible, I suggest that we replace them.

Then I would suggest using this instead:
```suggestion
        getRowIds: function() {
            return this.rows.keys();
        },
```
and let the caller convert to an array if they need to.